src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -72,17 +76,54 @@ export class RundownComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   private subscribeToEventObserver(): void {
-    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(event => this.scrollViewToSegment(event)))
-    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(event => this.scrollViewToSegment(event)))
+    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(event => this.prioritizedDebounce(event)))
+    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(event => this.prioritizedDebounce(event)))
   }
 
-  public scrollViewToSegment(event: PartEvent): void {
-    const element: HTMLElement | null = document.getElementById(event.segmentId)
-    if (!element || this.isInsideViewport(element)) {
+  private prioritizedDebounce(event: PartEvent): void {
+    this.startDebounceTimer()
+    const currentPriority: number = this.highestPriorityEvent ? this.getEventPriority(this.highestPriorityEvent) : 0
+    const eventPriority: number = this.getEventPriority(event)
+    if (eventPriority >= currentPriority) {
+      this.highestPriorityEvent = event
+    }
+  }
+
+  private startDebounceTimer(): void {
+    this.clearDebounceTimer()
+    this.debounceTimer = setTimeout(() => {
+      this.clearDebounceTimer()
+      if (!this.highestPriorityEvent) {
+        return
+      }
+      this.scrollViewToSegment(this.highestPriorityEvent.segmentId)
+      this.highestPriorityEvent = undefined
+    }, DEBOUNCE_EVENT_THRESHOLD_IN_MS)
+  }
+
+  private clearDebounceTimer(): void {
+    clearTimeout(this.debounceTimer)
+    this.debounceTimer = undefined
+  }
+
+  private getEventPriority(event: TypedEvent): number {
+    switch (event.type) {
+      case RundownEventType.SET_NEXT:
+        return 5
+      case RundownEventType.TAKEN:
+        return 4
+      default:
+        return 0
+    }
+  }
```

Comment:
Mostly it was to signify that out of the current available events we can listen to these are the most important ones out of all. But i see your point since it only pertains to this specific context at the moment!